### PR TITLE
chore: convert umi-utils to TypeScript

### PR DESCRIPTION
**Pull request by xiaohuoni**
##### Checklist

- [x] `npm test` passes
- [x] tests are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change
image
###### After
image

image
###### After
image

### CODE REVIEW

File: packages/umi-utils/src/chunkName.ts [hunk 1]
```diff
@@ -1,14 +1,19 @@
 import winPath from './winPath';
 
-function stripFirstSlash(path) {
+function stripFirstSlash(path: string) {
```
Comment (ycjcl868) [inline packages/umi-utils/src/chunkName.ts]: 返回值类型也要加下

File: packages/umi-utils/src/chunkName.ts [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
   }
 }
 
-export default function chunkName(cwd, path) {
+/**
+ * Get Relative Path
+ * @param cwd
+ * @param path
+ */
+export default function chunkName(cwd: string, path: string) {
```
Comment (ycjcl868) [inline packages/umi-utils/src/chunkName.ts]: 返回值

File: packages/umi-utils/src/compatDirname.ts [hunk 3]
```diff
@@ -2,7 +2,13 @@ import { dirname, join } from 'path';
...
 import winPath from './winPath';
 
-export default function(path, cwd, fallback) {
+/**
+ * Find module path
+ * @param path module name
+ * @param cwd process cwd
+ * @param fallback
+ */
+export default function(path: string, cwd: string, fallback?: any) {
```
Comment (ycjcl868) [inline packages/umi-utils/src/compatDirname.ts]: fallback `any`

File: packages/umi-utils/src/isUrl.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import isUrl from 'is-url';
+
+/**
+ * Check whether a string is a URL.
+ * @param path
+ */
+export default function(path: string) {
```
Comment (ycjcl868) [inline packages/umi-utils/src/isUrl.ts]: 返回值

File: packages/umi-utils/src/prettierFile.ts [hunk 5]
```diff
@@ -1,6 +1,11 @@
 import * as prettier from 'prettier';
 
-export default function(fileContent, parser = 'typescript') {
+/**
+ * prettier code
+ * @param fileContent file flow string
+ * @param parser default typescript
+ */
+export default function(fileContent: string, parser: string = 'typescript') {
```
Comment (ycjcl868) [inline packages/umi-utils/src/prettierFile.ts]: `parser` 是 literal type, `parser: 'typescript' | 'javascript' = 'typescript'`

File: packages/umi-utils/src/winEOL.ts [hunk 6]
```diff
@@ -4,7 +4,11 @@ const isWindows = typeof process !== 'undefined' && process.platform === 'win32'
 // 为了测试精准需要将换行符转化一下
 // https://github.com/cssmagic/blog/issues/22
 
-export default content => {
+/**
+ * Convert Windows crlf to lf (\r\n to \n)
+ * @param content
+ */
+export default (content: any) => {
```
Comment (ycjcl868) [inline packages/umi-utils/src/winEOL.ts]: any 后这里就没类型了

File: packages/umi-utils/src/winPath.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import slash from 'slash2';
+
+/**
+ * Convert Windows backslash paths to slash paths
+ * @param path
+ */
+export default function(path: string) {
```
Comment (ycjcl868) [inline packages/umi-utils/src/winPath.ts]: 返回值

File: packages/umi-utils/src/prettierFile.ts (hunk 5 above)
Comment (xiaohuoni) [reply packages/umi-utils/src/prettierFile.ts]: 这个其实不是，还有json和less等等

File: packages/umi-utils/src/compatDirname.ts (hunk 3 above)
Comment (xiaohuoni) [reply packages/umi-utils/src/compatDirname.ts]: 这里运行为空

File: packages/umi-utils/src/prettierFile.ts (hunk 5 above)
Comment (xiaohuoni) [reply packages/umi-utils/src/prettierFile.ts]: 我写全吧

**Comment by xiaohuoni** [comment]
https://github.com/umijs/umi/pull/3193

**Pull request merged**